backend/Onied/Tests.Users/UnitTests/ServiceTests/ProfileServiceTests.cs

```diff
@@ -0,0 +1,138 @@
+using System.Security.Claims;
+using AutoFixture;
+using AutoMapper;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Moq;
+using Tests.Users.UnitTests.Helpers;
+using Users.Data.Entities;
+using Users.Dtos.Profile.Request;
+using Users.Dtos.Users.Response;
+using Users.Profiles;
+using Users.Services.ProfileProducer;
+using Users.Services.ProfileService;
+
+namespace Tests.Users.UnitTests.ServiceTests;
+
+public class ProfileServiceTests
+{
+    private readonly ProfileService _service;
+    private readonly Fixture _fixture = new();
+    private readonly Mock<IProfileProducer> _profileProducer = new();
+    private readonly IMapper _mapper = new Mapper(new MapperConfiguration(cfg => cfg.AddProfile(new AppMappingProfile())));
+    private readonly Mock<TestUserManager> _userManager = new();
+
+    public ProfileServiceTests()
+    {
+        _service = new ProfileService(_profileProducer.Object, _userManager.Object, _mapper);
+    }
+
+    [Fact]
+    public async Task Get_ReturnsUnauthorized_WhenCannotGetUser()
+    {
+        // Arrange
+        _userManager.Setup(manager => manager.GetUserAsync(It.IsAny<ClaimsPrincipal>())).ReturnsAsync(null as AppUser);
+
+        // Act
+        var result = await _service.Get(new ClaimsPrincipal());
+
+        // Assert
+        Assert.IsType<UnauthorizedHttpResult>(result);
+    }
+
+    [Fact]
+    public async Task Get_ReturnsOk_WhenUserExists()
+    {
+        // Arrange
+        var user = _fixture.Create<AppUser>();
+        _userManager.Setup(manager => manager.GetUserAsync(It.IsAny<ClaimsPrincipal>())).ReturnsAsync(user);
+
+        // Act
+        var result = await _service.Get(new ClaimsPrincipal());
+
+        // Assert
+        Assert.IsType<Ok<UserProfileResponse>>(result);
+        var profile = (result as Ok<UserProfileResponse>)!.Value!;
+        Assert.Equal(user.FirstName, profile.FirstName);
+        Assert.Equal(user.LastName, profile.LastName);
+        Assert.Equal(user.Gender, profile.Gender);
+        Assert.Equal(user.Avatar, profile.Avatar);
+        Assert.Equal(user.Email, profile.Email);
```

Comment:
Можно заменить
```suggestion
        Assert.Equivalent(profile, user)
```

Проверяет по всем публичным свойствам, что есть в expected. В данном случае поменял expected и actual местами, т.к. в user больше полей, чем в ответе. Можно промаппить ручками, можно оставить так. Да, семантически это будет не совсем правильно, зато короче.

Конкретно здесь на твое усмотрение.